modules/renter/streamcache_test.go

```diff
@@ -0,0 +1,135 @@
+package renter
+
+import (
+	"container/heap"
+	"reflect"
+	"strconv"
+	"testing"
+	"time"
+)
+
+// TestHeapImplementation tests that the streamCache heap functions properly
+func TestHeapImplementation(t *testing.T) {
+	// Initializing minimum variables
+	sc := new(streamCache)
+	sc.Init()
+
+	// Testing Push to Heap
+	length := len(sc.streamHeap)
+	cd := &chunkData{
+		id:         "Push",
+		data:       []byte{},
+		lastAccess: time.Now(),
+	}
+	heap.Push(&sc.streamHeap, cd)
+
+	// Confirming the length of the heap increased by 1
+	if len(sc.streamHeap) != length+1 {
+		t.Error("Length of heap did not change, chunkData was not pushed onto Heap. Length of heap is still ", len(sc.streamHeap))
+	}
+	// Confirming the chunk added was the one expected
+	if !reflect.DeepEqual(cd, sc.streamHeap[0]) {
+		t.Error("Chunk on top of heap is not the chunk that was just pushed on, chunkData.id =", sc.streamHeap[0].id)
+	}
+
+	// Add more chunks to heap
+	for i := 0; i < 3; i++ {
+		heap.Push(&sc.streamHeap, &chunkData{
+			id:         strconv.Itoa(i),
+			data:       []byte{},
+			lastAccess: time.Now(),
+		})
+	}
+
+	// Testing Heap update
+	// Confirming recently accessed elements get moved to the bottom of Heap
+	cd = sc.streamHeap[0]
+	sc.streamHeap.update(cd, cd.id, cd.data, time.Now())
+	if reflect.DeepEqual(cd, sc.streamHeap[0]) {
+		t.Error("Heap order was not updated. Recently accessed element not at bottom of heap")
```

Comment:
I don't think this check is necessary since it only proves that the new element doesn't end up as the root of the heap. If you still want to keep it you should update the error message though.